Add upnp sensors for IP, Status, and Uptime

ehendrix23 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add 3 additional sensors to provide IP address, Status, and Uptime

Note, PR based on changes from PR #54489
## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [X] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

ehendrix23 committed:
New binary sensor for connectivity

ehendrix23 committed:
New binary sensor for connectivity

ehendrix23 committed:
Add binary_sensor

ehendrix23 committed:
Add binary_sensor

ehendrix23 committed:
Merge branch 'home-assistant:dev' into upnp_connectivity_binary_sensor

ehendrix23 committed:
Merge branch 'upnp_connectivity_binary_sensor' of https://github.com/ehendrix23/home-assistant into upnp_connectivity_binary_sensor

ehendrix23 committed:
Handle values returned as None

ehendrix23 committed:
Small text update for Uptime

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Updates based on review

ehendrix23 committed:
Updates based on review

ehendrix23 committed:
Merge branch 'upnp_connectivity_binary_sensor' of https://github.com/ehendrix23/home-assistant into upnp_connectivity_binary_sensor

ehendrix23 committed:
Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

ehendrix23 committed:
Further updates based on review

ehendrix23 committed:
Merge branch 'upnp_connectivity_binary_sensor' of https://github.com/ehendrix23/home-assistant into upnp_connectivity_binary_sensor

ehendrix23 committed:
Set device_class as a class atribute

ehendrix23 committed:
Set device_class as a class atribute

ehendrix23 committed:
Create 1 combined data coordinator
and UpnpEntity class

ehendrix23 committed:
Create 1 combined data coordinator
and UpnpEntity class

ehendrix23 committed:
Updates on coordinator

ehendrix23 committed:
Updates on coordinator

ehendrix23 committed:
Update comment

ehendrix23 committed:
Fix in async_step_init for coordinator

ehendrix23 committed:
Add async_get_status to mocked device
and set times polled for each call seperately

ehendrix23 committed:
Updated to get device through coordinator
Check polling for each status call seperately

ehendrix23 committed:
Use collections.abc instead of Typing for Mapping

ehendrix23 committed:
Remove adding device to hass.data as coordinator
is now saved

ehendrix23 committed:
Removed setting _coordinator

ehendrix23 committed:
Removed setting _coordinator

ehendrix23 committed:
Added myself as codeowner

ehendrix23 committed:
Update type in __init__

ehendrix23 committed:
Removed attributes from binary sensor

ehendrix23 committed:
Fix async_unload_entry

ehendrix23 committed:
Add expected return value to is_on

ehendrix23 committed:
Add additional UPNP sensors

ehendrix23 committed:
Add additional UPNP sensors

ehendrix23 committed:
Add unit seconds for Uptime

ehendrix23 committed:
Add unit seconds for Uptime

ludeeus commented:
Converted this to draft for now as it uses elements not yet merged (from #54489).

ehendrix23 committed:
Add light platform to MyQ (#54611)

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Upgrade pre-commit to 2.14.0 (#54719)

bdraco committed:
Convert nmap_tracker to be a config flow (#54715)

scop committed:
Upgrade mypy to 0.910 and types-* (#54574)

Co-authored-by: Daniel Hjelseth Høyer <mail@dahoiv.net>

cdce8p committed:
Use EntityDescription - qbittorrent (#54428)

cdce8p committed:
Use EntityDescription - point (#54363)

janiversen committed:
Activate mypy in sabnzbd (#54539)

scop committed:
Avoid some implicit generic Anys (#54577)

janiversen committed:
Active mypy for netio (#54543)

Jc2k committed:
Add device classes that were part of deprecated air quality entity (#54075)

mib1185 committed:
Use state class total increasing for TPLink smart plugs (#54723)

bdraco committed:
Bump zeroconf to 0.36.0 (#54720)

emontnemery committed:
Remove last_reset attribute from FritzBoxEnergySensor (#54644)

cdce8p committed:
Use EntityDescription - logi_circle (#54429)

janiversen committed:
Activate mypy for sesame (#54546)

cdce8p committed:
Use EntityDescription - mitemp_bt (#54503)

mib1185 committed:
Add DEVICE_CLASS_UPDATE to Binary Sensor (#53945)

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Commit:
[ci skip] Translation update

Bert-R committed:
Fix Google Calendar event loading (#54231)

ludeeus committed:
Add device class update to the updater binary_sensor (#54732)

ludeeus committed:
Add device class update to hassio update entities (#54733)

frenck committed:
Update deCONZ to use new state classes (#54729)

andre-richter committed:
Add state classes to Vallox sensors (#54297)

janiversen committed:
Activate mypy for google_pubsub (#54649)

ppetru committed:
Energy support for Solax inverters (#54654)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bieniu committed:
Use PM1, PM25 and PM10 device classes in Nettigo Air Monitor integration (#54741)

bieniu committed:
Use DEVICE_CLASS_UPDATE in Shelly integration (#54746)

ludeeus committed:
Use entity descriptions for hassio entities (#54749)

raman325 committed:
Relax zwave_js lock discovery rules to cover more use cases (#54710)

frenck committed:
DSMR: Remove icon from sensors with gas device class (#54752)

nielstron committed:
Bump pyfronius version to 0.5.5 (#54758)

- allows for trailing slashes in configuration (which would otherwise cause errors in the newest fronius firmware)
- fixes units of energy related sensors

squishykid committed:
Solax 0.2.8 (#54759)

frenck committed:
Toon: Remove icon from sensors with gas device class (#54753)

Jc2k committed:
Deprecate homekit_controller's air quality entity in favor of separate sensor entities (#54673)

samsinnamon committed:
Airtouch4 integration (#43513)

* airtouch 4 climate control integration

* enhance tests for airtouch. Fix linting issues

* Fix tests

* rework tests

* fix latest qa issues

* Clean up

* add already_configured message

* Use common string

* further qa fixes

* simplify airtouch4 domain storage

Co-authored-by: Franck Nijhof <git@frenck.dev>
Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Add zwave_js Protection CC select entities (#54717)

* Add Protection CC select entities

comment

* Disable entity by default

* use class attribute

* Enable protection entity by default

* add guard for none

mib1185 committed:
Use DEVICE_CLASS_UPDATE in Synology DSM (#54769)

frenck committed:
Use path helper method for principal file in google_pubsub (#54744)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

ehendrix23 committed:
Add upnp binary sensor for connectivity status (#54489)

* New binary sensor for connectivity

* Add binary_sensor

* New binary sensor for connectivity

* Add binary_sensor

* Handle values returned as None

* Small text update for Uptime

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Updates based on review

* Update homeassistant/components/upnp/binary_sensor.py

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

* Further updates based on review

* Set device_class as a class atribute

* Create 1 combined data coordinator
and UpnpEntity class

* Updates on coordinator

* Update comment

* Fix in async_step_init for coordinator

* Add async_get_status to mocked device
and set times polled for each call seperately

* Updated to get device through coordinator
Check polling for each status call seperately

* Use collections.abc instead of Typing for Mapping

* Remove adding device to hass.data as coordinator
is now saved

* Removed setting _coordinator

* Added myself as codeowner

* Update type in __init__

* Removed attributes from binary sensor

* Fix async_unload_entry

* Add expected return value to is_on

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

StevenLooman commented:
I'm removing the draft-status, as #54489 is merged.

StevenLooman commented:
Can you rebase the last two commits (those seem to be new compared to your previous PR), @ehendrix23?

janiversen committed:
set common test entity name. (#54697)

Danielhiversen committed:
Tractive, update library (#54775)

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

ehendrix23 committed:
Rebase issue fix

ehendrix23 committed:
Merge branch 'upnp_additional_sensors' of https://github.com/ehendrix23/home-assistant into upnp_additional_sensors

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#54718).

Once you do that we will be able to review and accept this pull request.

Thanks!

ehendrix23 commented:
Closing as clearly I messed up my rebase.
